### PR TITLE
Channel Based Class to support gnocchi, designate-bind and glance-simplestreams

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- removed is_os_channel_based from base class

Closes: #94, #106, #125

### CODE REVIEW

**Review by rgildein — APPROVED** [review]
LGTM

File: cou/apps/channel_based.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    def post_upgrade_plan(self, target: OpenStackRelease) -> list[UpgradeStep]:
+        """Post Upgrade planning.
+
+        Wait until the application reaches the idle state and then check the target workload.
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: Plan that will add post upgrade as sub steps.
+        :rtype: list[UpgradeStep]
+        """
+        if self.is_versionless:
```
Comment (agileshaw) [inline cou/apps/channel_based.py:72]: Can we include this check in the docstring?

File: cou/apps/base.py [hunk 2]
```diff
@@ -184,6 +184,11 @@ def channel(self) -> str:
         :return: Charm channel. E.g: ussuri/stable
         :rtype: str
         """
+        # NOTE(gabrielcocenza) Some applications current_os_release points
+        # to channel_codename without setting the channel before. In that
+        # case we check if the attribute is already set.
```
Comment (agileshaw) [inline cou/apps/base.py:189]:
```suggestion
        # NOTE(gabrielcocenza) Some applications current_os_release points
        # to channel_codename without setting the channel before. In that
        # case we set the attribute before accessing it.
```

**Review by agileshaw — CHANGES_REQUESTED** [review]
Overall LGTM. Just a few small changes required inline

**Review by agileshaw — APPROVED** [review]
LGTM, thanks

**Review by rgildein — APPROVED** [review]
LGTM

**Pull request merged**